server/swap/swap.go

```diff
@@ -1639,7 +1647,7 @@ func (s *Swapper) processInit(msg *msgjson.Message, params *msgjson.Init, stepIn
 	actor.status.mtx.Unlock()
 
 	stepInfo.match.mtx.Lock()
-	stepInfo.match.Status = stepInfo.nextStep
+	stepInfo.match.Status = stepInfo.nextStep // handleInit (gate mechanism) won't allow backward progress
 	stepInfo.match.mtx.Unlock()
```

Comment:
This is just to clarify what "handleInit gating" mechanism actually is (it is mentioned couple of lines above),

I read through the code to verify that it indeed works, but could also add a unit-test to cover this scenario if you want.